src/com/google/common/css/MultipleMappingSubstitutionMap.java

```diff
@@ -76,5 +77,5 @@ public static ValueWithMappings createForSingleMapping(String key, String value)
    * associated with this substitution. These mappings are available as
    * {@link ValueWithMappings#mappings}.
    */
-  public ValueWithMappings getValueWithMappings(String key);
+  public ValueWithMappings getValueWithMappings(String key, @Nullable Map<String, String> inputMappings);
```

Comment:
Oh, I see. I'd forgotten that you might need to know the existing mappings to avoid collisions. `MinimalSubstitutionMap` already has the `outputValueBlacklist`, as you point out, so I think that's a reasonable approach.